hips/hip-XXXX.md

```diff
@@ -0,0 +1,130 @@
+---
+hip: 9999
+title: "registries.conf support for OCI registry management"
+authors: [ "George Jenkins <gvjenkins@gmail.com>", "Andy Block <ablock@redhat.com>" ]
+created: "2025-02-16"
+type: "feature"
+status: "draft"
+---
+
+## Abstract
+
+[registries.conf][registries-conf] is a an alternative specification (to Docker's `docker/config.json`) for managing OCI registry configuration. That supports more advanced features, and has been standardized by other container ecosystem CLI tools (podman, buildah, skopeo, etc)
+
+Support would extend Helm's flexilbity with respect to OCI configuration, including repository prefixes and aliases. This HIP focuces on the implementation of supporting `registries.conf`. Further HIPs to be created for exposing functionality based on utilzing `registries.conf`.
+
+[registries-conf]: <https://github.com/containers/image/blob/main/docs/containers-registries.conf.5.md> "registries.conf specification"
+
+## Motivation
+
+`registries.conf` provides much more flexibible support for OCI registry management than the current Docker CLI based configuration format that Helm uses today.
+
+Including support for repository prefixes (allowing different credentials for different prefixes), registry mirrors, and registry "aliasing". Features Helm would like to introduce. But is currenly blocked by a lack of mechanism to store detail
+
+(in particular, the existing registry configuration Helm uses, Docker’s `$HOME/.docker/config.json`, etc do not support registry aliases nor prefixes)
+
+## Rationale
+
+Utilizing an existing specification / libraries enables Helm immediate build upon an existing standard.
+Rather than Helm inventing its own convention.
+
+`registries.conf` was picked as being a format intended for consumption beyond the Docker application container ecosystem.
+
+## Specification
+
+Helm will utilize the `registries.conf` specification when determinging OCI registry information (authentication credentials, etc):
+<https://github.com/containers/image/blob/main/docs/containers-registries.conf.5.md>
+
+
+`registries.conf` will be preferred, either when a `registries.conf` file already exists on the users system.
+Or when Helm supports and a user utilizes functionality that can not be supported by Docker's comnfigurtation file in the future.
+
+<!-- The usage of `registries.conf` will be forced, when in the future -->
+
+For example, a registry login command:
+```bash
+helm registry login "oci.example.com" --username foo --password bar
+```
+
+Will result in the configration exerpt (if, and only if, `registries.conf` exists on the users local system):
+```toml
+# registries.conf
+[[registry]]
+prefix = "oci.example.com"
+```
+
+```json
+# auth.json
+"auths": {
+  ...
+  "oci.example.com": {
+    "auth": "Zm9vOmJhcgo="
```

Comment:
I was testing how podman stores the credentials and it appears they are base64 encoded. With the current Docker based system is can (and you should) use a credential helper that enables storing in a secure manner (the systems credential store). For example, this stores it in the keychain on OS X or something like pass on Linux.

How can this security degradation be mitigated?

Reply:
The oras-go confix.json credentials backend uses the native store https://github.com/oras-project/oras-go/tree/main/registry/remote/credentials

If ORAS was to support this, I would expect something like that.

Reply:
@mattfarina [Credential Helpers](https://github.com/containers/image/blob/main/docs/containers-registries.conf.5.md#global-settings) can be leveraged natively with Podman as well as in ORAS as @TerryHowe shared